Cubical/Algebra/CommMonoid/GrothendieckGroup.agda

```diff
@@ -0,0 +1,228 @@
+{-
+Module in which the Grothendieck Group ("Groupification") of a commutative monoid is defined.
+-}
+{-# OPTIONS --safe #-}
+module Cubical.Algebra.CommMonoid.GrothendieckGroup where
+
+open import Cubical.Algebra.CommMonoid.Base
+open import Cubical.Algebra.CommMonoid.CommMonoidProd
+
+open import Cubical.Algebra.AbGroup.Base
+open import Cubical.Algebra.Group.Base
+open import Cubical.Algebra.Group.Properties
+open import Cubical.Algebra.Group.Morphisms
+open import Cubical.Algebra.Monoid.Base
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Structure
+
+open import Cubical.HITs.SetQuotients.Base
+open import Cubical.HITs.SetQuotients.Properties
+
+open import Cubical.Relation.Binary.Base
+
+
+private
+  variable
+    ℓ ℓ' : Level
+
+
+module _ (M : CommMonoid ℓ) where
+  open BinaryRelation
+
+  M² : CommMonoid _
+  M² = CommMonoidProd M M
+
+  open CommMonoidStr ⦃...⦄
+  private
+    instance
+      _ = snd M
+      _ = snd M²
+
+  R : ⟨ M² ⟩  → ⟨ M² ⟩ → Type _
+  R (a₁ , b₁) (a₂ , b₂) =  Σ[ k ∈ ⟨ M ⟩ ] k · (a₁ · b₂) ≡ k · (b₁ · a₂)
+
+  M²/R : Type _
+  M²/R = ⟨ M² ⟩ / R
+
+  0/R : M²/R
+  0/R = [ ε , ε ]
+
+  private
+    _+/_ : M²/R → M²/R → M²/R
+    x +/ y = setQuotBinOp isReflR isReflR _·_ isCongR x y
+      where
+      isReflR : isRefl R
+      isReflR (a , b) = ε , cong (ε ·_) (comm a b)
+
+      isCongR : ∀ u u' v v' → R u u' → R v v' → R (u · v) (u' · v')
+      isCongR (a₁ , b₁) (a₂ , b₂) (a₃ , b₃) (a₄ , b₄) (k , p) (s , q) = k · s , proof
+        where
+        proof =
+          (k · s) · ((a₁ · a₃) · (b₂ · b₄)) ≡⟨ lemma ⟩
+          (k · (a₁ · b₂)) · (s · (a₃ · b₄)) ≡⟨ cong₂ _·_ p q ⟩
+          (k · (b₁ · a₂)) · (s · (b₃ · a₄)) ≡⟨ sym lemma ⟩
+          (k · s) · ((b₁ · b₃) · (a₂ · a₄)) ∎
+            where
+            Assoc : ∀ {x y z} → x · (y · z) ≡ (x · y) · z
```

Comment:
Capitalization of a term is against the style rules... So it would be good to change it to something like assocM2/R

Reply:
Let's hope I got it right this time :)